<launch>
<arg name="0_robot"  default="robot_0"/>
<arg name="1_robot"  default="robot_1"/>
<arg name="2_robot"  default="robot_2"/>
<arg name="3_robot"  default="robot_3"/>
<arg name="4_robot"  default="robot_4"/>

          <group ns="/robot_0/map_merge">
          <param name="init_pose_x" value="-2"/>
          <param name="init_pose_y" value="0"/>
          <param name="init_pose_z" value=" 0.0"/>
          <param name="init_pose_yaw"   value="0"/>
          </group>
          <group ns="/robot_1/map_merge">
          <param name="init_pose_x" value="-1"/>
          <param name="init_pose_y" value="0"/>
          <param name="init_pose_z" value=" 0.0"/>
          <param name="init_pose_yaw"   value="0"/>
          </group>
          <group ns="/robot_2/map_merge">
          <param name="init_pose_x" value="0"/>
          <param name="init_pose_y" value="0"/>
          <param name="init_pose_z" value=" 0.0"/>
          <param name="init_pose_yaw"   value="0"/>
          </group>
          <group ns="/robot_3/map_merge">
          <param name="init_pose_x" value="1"/>
          <param name="init_pose_y" value="0"/>
          <param name="init_pose_z" value=" 0.0"/>
          <param name="init_pose_yaw"   value="0"/>
          </group>
          <group ns="/robot_4/map_merge">
          <param name="init_pose_x" value="-2"/>
          <param name="init_pose_y" value="-0.5"/>
          <param name="init_pose_z" value=" 0.0"/>
          <param name="init_pose_yaw"   value="0"/>
          </group>
	<node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge" output="screen">
      <param name="robot_map_topic" value="map"/>
      <param name="robot_namespace" value=""/>
      <param name="merged_map_topic" value="merged_map"/>
      <param name="world_frame" value="/map"/>
      <param name="known_init_poses" value="true"/>
      <param name="merging_rate" value="15"/>
      <param name="discovery_rate" value="0.05"/>
      <param name="estimation_rate" value="0.5"/>
      <param name="estimation_confidence" value="1.0"/>
    </node>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg 0_robot)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg 0_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg 1_robot)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg 1_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg 2_robot)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg 2_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg 3_robot)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg 3_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg 4_robot)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg 4_robot)/map 200"/>
</launch>